vms/avm/vm.go

```diff
@@ -421,22 +452,48 @@ func (vm *VM) Linearize(_ context.Context, stopVertexID ids.ID, toEngine chan<-
 		vm.txBackend,
 		vm.chainManager,
 		&vm.clock,
-		vm.mempool,
+		mempool,
 	)
 
-	vm.network = network.New(
+	vm.network, err = network.New(
 		vm.ctx,
 		vm.parser,
-		vm.chainManager,
-		vm.mempool,
+		network.NewLockedTxVerifier(
+			&vm.ctx.Lock,
+			vm.chainManager,
+		),
+		mempool,
 		vm.appSender,
+		vm.registerer,
+		txGossipHandlerID,
+		maxValidatorSetStaleness,
+		txGossipMaxGossipSize,
+		txGossipPollSize,
+		txGossipThrottlingPeriod,
+		txGossipThrottlingLimit,
+		txGossipBloomMaxExpectedElements,
+		txGossipBloomFalsePositiveProbability,
+		txGossipBloomMaxFalsePositiveProbability,
 	)
+	if err != nil {
+		return fmt.Errorf("failed to initialize network: %w", err)
+	}
+
+	// Notify the network of our current peers
+	for nodeID, version := range vm.connectedPeers {
+		if err := vm.network.Connected(ctx, nodeID, version); err != nil {
+			return err
+		}
+	}
+	vm.connectedPeers = nil
 
 	// Note: It's important only to switch the networking stack after the full
 	// chainVM has been initialized. Traffic will immediately start being
 	// handled asynchronously.
 	vm.Atomic.Set(vm.network)
 
+	go vm.network.Gossip(context.TODO(), txGossipFrequency)
```

Comment:
nit: I feel like we do have/should already have a long-lived vm context at this point?

Reply:
This is the first goroutine that we are creating in the AVM (except for the state pruner immediately below which we'll remove after `v1.11.x` activates.

Reply:
So, I guess I'll add it for this haha